### PR TITLE
fix: `FullyQualifiedStrictTypesFixer` - reset cache even if there is no `use`

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/8182

Even though the test case looks like an edge case, it covers the issue: take a look at the 1st commit and its CI - when file `A.php` runs before `B.php` it fails, when the order is swapped, all is fine.

### CODE REVIEW

**Review by SpacePossum — APPROVED** [review]

**Comment by mvorisek** [comment]
Thank you, finally, to honoring my comment - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7705#discussion_r1710359312 !

**Comment by Wirone** [comment]
@mvorisek I believe this could be fixed in a original PR and wouldn't introduce an issue, but it would be just easier if you provided context back then (test case, snippet etc). Without this I did not take this comment into consideration, because I asked for explanation twice and did not get it 🤷. It looks like you were right in the end, I have no problem with admitting it, I just like to understand _why_ I need to change my code, and it wasn't the case then.

File: src/Fixer/Import/FullyQualifiedStrictTypesFixer.php [hunk 1]
```diff
@@ -409,9 +409,10 @@ private function refreshUsesCache(array $uses): void
 
         $this->cacheUsesLast = $uses;
```
Comment (mvorisek) [inline src/Fixer/Import/FullyQualifiedStrictTypesFixer.php:411]:
```suggestion
```

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you @kubawerlos 🍻!
